### ISSUE TITLE
CHIPIR: IOC and OPI for Collimator

### ISSUE THREAD

**Issue by kjwoodsISIS** [ISISComputingGroup/IBEX#6218]
As a CHIPIR scientist I want to be able to control the beamline collimator & jaws in experiments. 

This device is galil based, but uses some custom galil code that runs on the controller. This is similar to the ORCs but much simpler in operation - it uses switches to determine positions (as well as a secondary encoder but this may or may not work still)

What we need to do is treat this like a stream device more than a galil axis, as the code should still run on the controller. the .gal code is available in `C:\labview modules\Instruments\CHIPIR\CHIPIR Collimator` along with the VI, which is shown below.

The VI also includes the jaws, so we should add this in the OPI - we can just embed a single jawset OPI which already exists, but we should make this configurable via a macro (ie. so you can use JAWS1, JAWS2 etc) 

#### Acceptance Criteria
1. Using the IBEX client, I can control the beamline collimator & jaws
1. Using genie_python, I can control the beamline collimator & jaws 

#### Notes
1. The CHIPIR collimator and jaws should be controlled from a common OPI
1. The collimator and jaws are Galil controlled.
1. Check the [CHIPIR Instrument wiki](https://github.com/ISISComputingGroup/IBEX/wiki/CHIPIR-Instrument-Details).
1. The OPI must offer functionality equivalent to the existing VI (located here: `C:\LabVIEW Modules\Instruments\CHIPIR\CHIPIR Collimator`).
   * @davidkeymer  did the original LabVIEW implementation. Consult @davidkeymer  for details.

![image](https://github.com/ISISComputingGroup/IBEX/assets/14823767/10ea3c6c-dc6e-41ab-b5e5-02891f767144)

time in planning 00:52 23/5/24

**Comment by rerpha** [comment]
think for this ticket i am going to: 

redesign OPI from scratch 
use ORC-like approach for galil code - load code onto controller on startup as we do, then use custom motorextensions db for logic etc. - this means we can read the encoder or use the galil for other stuff if needed (though still nto sure if they are using it for any other axes, dont think they are)

**Comment by rerpha** [comment]
https://github.com/ISISComputingGroup/EPICS-galil/pull/76

**Comment by rerpha** [comment]
"home all" on the VI will be achieved by an instrument script like so:

```

MTR = "MOT:MTR01"

def home_all():
    g.set_pv(f"{MOT}01.HOMF", 1, is_local=True)
    g.waitfor_move()
    g.set_pv(f"{MOT}02.HOMF", 1, is_local=True)
    g.waitfor_move()
    g.set_pv(f"{MOT}03.HOMF", 1, is_local=True)
    g.set_pv(f"{MOT}04.HOMF", 1, is_local=True)
    g.set_pv(f"{MOT}05.HOMF", 1, is_local=True)
    g.set_pv(f"{MOT}06.HOMF", 1, is_local=True)


```

**Comment by KathrynBaker** [comment]
That looks like it homes 1, then it homes 2, then it homes 3,4,5, and 6 at the same time. Which homing routines it uses may also have an impact on that, as it certainly is treating axis 1 and 2 differently.

**Comment by rerpha** [comment]
correct - that's what the VI did as well

**Comment by rerpha** [comment]
just confirming but the galil does in fact control the jaws and jaws in/out axis - this isn't a problem really as the code will be run on separate threads to our homing thread.

**Comment by Chsudeepta** [comment]
All the changes have been reviewed and tested
